apps/editing-toolkit/editing-toolkit-plugin/dotcom-fse/plugins/editor-template-classes/index.js

```diff
@@ -23,21 +24,26 @@ const EditorTemplateClasses = withSelect( ( select ) => {
 	} );
 	return { templateClasses };
 } )( ( { templateClasses } ) => {
-	const blockListInception = setInterval( () => {
-		const blockListParent = document.querySelector( '.block-editor-writing-flow' );
+	useEffect( () => {
+		const blockListInception = setInterval( () => {
+			const blockListParent = document.querySelector(
+				'.editor-styles-wrapper > .block-editor-block-list__layout'
+			);
 
-		if ( ! blockListParent ) {
-			return;
-		}
-		clearInterval( blockListInception );
+			if ( ! blockListParent ) {
+				return;
+			}
+			clearInterval( blockListInception );
 
-		blockListParent.className = classNames(
-			'block-editor-writing-flow',
-			'a8c-template-editor fse-template-part',
-			...templateClasses
-		);
-	} );
+			blockListParent.className = classNames(
+				blockListParent.className,
+				'a8c-template-editor fse-template-part',
```

Comment:
Even after moving to useEffect it still may be worth double checking if these classes already exist on the element before adding them, as it looks like we are still getting duplicates. 🤔

Reply:
The template classes dependency is `undefined` until data about the template part is fetched, which is why we get a set of duplicates (one instance of set interval is called for the `undefined` value, and another for the template class value of `header` or `footer`)

[In my previous example ](https://github.com/Automattic/wp-calypso/pull/55109#discussion_r682013675), I included a condition to check for an `undefined` template class to guard against this. Addie's approach definitely works too.